service/worker/service.go

```diff
@@ -327,68 +346,169 @@ func (svc *Service) stepDown() {
+				SubcriptionKey: new.SubscriptionID,
+				Broker:         svc.workerConfig.Broker.Variant,
+				BrokerConfig:   svc.workerConfig.Broker.BrokerConfig,
+			}
+			svc.workgrp.Go(handler.Start)
+			svc.pushHandlers[new.Key()] = handler
+		}
+	}
+
+	svc.nodebindingCache = newBindings
+	return nil
+}
+
+func (svc *Service) handleSubUpdates(newSubsPairs []registry.Pair) error {
 	oldSubs := svc.subCache
+	newSubs := []subscription.Model{}
+
+	for _, pair := range newSubsPairs {
+		var sub subscription.Model
+		err := json.Unmarshal(pair.Value, &sub)
+		if err != nil {
+			return err
+		}
+		newSubs = append(newSubs, sub)
+	}
+
 	for _, old := range oldSubs {
 		found := false
 		for _, new := range newSubs {
-			if old == new {
+			if old.Key() == new.Key() {
 				found = true
 				break
 			}
 		}
 
 		if !found {
-			logger.Ctx(svc.ctx).Infow("sub removed", "sub_key", old)
+			logger.Ctx(svc.ctx).Infow("sub removed", "key", old.Key())
+			for _, nodebinding := range svc.nodebindingCache {
+				if nodebinding.SubscriptionID == old.Key() {
+					svc.nodeBindingCore.DeleteNodeBinding(svc.ctx, &nodebinding)
+				}
+			}
 		}
 	}
 
 	for _, new := range newSubs {
 		found := false
 		for _, old := range oldSubs {
-			if old == new {
+			if old.Key() == new.Key() {
 				found = true
 				break
 			}
 		}
 
 		if !found {
-			logger.Ctx(svc.ctx).Infow("sub added", "sub_key", new)
+			logger.Ctx(svc.ctx).Infow("sub added", "key", new.Key())
+			nb, err := svc.scheduler.Schedule(new, svc.nodebindingCache, svc.nodeCache)
+			if err != nil {
+				return err
+			}
+
+			err = svc.nodeBindingCore.CreateNodeBinding(svc.ctx, nb)
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	svc.subCache = newSubs
 	return nil
 }
 
-func (svc *Service) handleNodeUpdates(newNodes []string) error {
+func (svc *Service) handleNodeUpdates(newNodePairs []registry.Pair) error {
 	oldNodes := svc.nodeCache
+	newNodes := []node.Model{}
+
+	for _, pair := range newNodePairs {
+		var node node.Model
+		err := json.Unmarshal(pair.Value, &node)
+		if err != nil {
+			return err
+		}
+		newNodes = append(newNodes, node)
+	}
 
 	for _, old := range oldNodes {
 		found := false
 		for _, new := range newNodes {
-			if old == new {
+			if old.Key() == new.Key() {
 				found = true
 				break
 			}
 		}
 
 		if !found {
-			logger.Ctx(svc.ctx).Infow("node removed", "node_key", old)
+			logger.Ctx(svc.ctx).Infow("node removed", "key", old.Key())
+
+			for _, nodebinding := range svc.nodebindingCache {
+				if nodebinding.NodeID == old.ID {
+					svc.nodeBindingCore.DeleteNodeBinding(svc.ctx, &nodebinding)
+				}
+			}
+
 		}
 	}
 
 	for _, new := range newNodes {
 		found := false
 		for _, old := range oldNodes {
-			if old == new {
+			if old.Key() == new.Key() {
 				found = true
 				break
 			}
 		}
 
 		if !found {
-			logger.Ctx(svc.ctx).Infow("node added", "node_key", new)
+			logger.Ctx(svc.ctx).Infow("node added", "key", new.Key())
+			// Do nothing for now
```

Comment:
This should also trigger scheduler right?

Reply:
for now, rebalance is not implemented, we dont need to change the current assignment for node additions